src/neo-vm/ExecutionEngine.cs

```diff
@@ -866,6 +866,25 @@ private void ExecuteInstruction()
                         Push(Pop().GetInteger().Sqrt());
                         break;
                     }
+                case OpCode.MODMUL:
+                    {
+                        var modulus = Pop().GetInteger();
+                        var x2 = Pop().GetInteger();
+                        var x1 = Pop().GetInteger();
+                        Push(x1 * x2 % modulus);
```

Comment:
```suggestion
                        Push((x1 % modulus) * (x2 % modulus) % modulus);
```
Since `(a x b) mod m = ((a mod m) x (b mod m)) mod m`,  maybe this is much faster when modulus is small.

Reply:
What if `modulus` is larger?

Reply:
In any case, the memory consumed by `x1` and `modulus`, lets say `size(x1)` and `size(modulus)` will always be such that: `size(x1 % modulus) <= size(x1)`, and same applies to `x2`... when `modulus > x1`, then `size(x1 % modulus) == size(x1)`. So, this approach only limits memory on x1 and x2, not on modulus (that must be system limited to `2^256`). I guess the worst case here, that should be a test, is `MODMUL  (2^256-1)  x (2^256-1) % (2^256). In this case, the mod strategy is ineffective, thus multiplication of (2^256-1)  x (2^256-1) will carry on (and consume max ~ 512 bits = 64 bytes) and be reduced to 32 bytes again after mod.
If modulus is small or median on average, what I am assuming here, this strategy will help in terms of memory (although costing slightly more on computing).

Reply:
@erikzhang after a while thinking, maybe it's best to keep it the simplest. In the best case, it will save some computation and some memory... in worst case, it will cost more computing while maintaining upper bound of ~512 bits. Just let it as is.